test/unit/io/alignment_file/alignment_file_format_test_template.hpp

```diff
@@ -486,34 +487,35 @@ TYPED_TEST_P(alignment_file_write, write_empty_members)
 
 TYPED_TEST_P(alignment_file_write, default_options_all_members_specified)
 {
-    detail::alignment_file_output_format<TypeParam> format;
-
     std::ostringstream ostream;
 
-    alignment_file_header header{std::vector<std::string>{this->ref_id}};
-    header.ref_id_info.push_back({this->ref_seq.size(), ""});
-    header.ref_dict[this->ref_id] = 0;
+    {
+        detail::alignment_file_output_format<TypeParam> format{ostream};
 
-    this->tag_dicts[0]["NM"_tag] = 7;
-    this->tag_dicts[0]["AS"_tag] = 2;
-    this->tag_dicts[1]["xy"_tag] = std::vector<uint16_t>{3,4,5};
+        alignment_file_header header{std::vector<std::string>{this->ref_id}};
+        header.ref_id_info.push_back({this->ref_seq.size(), ""});
```

Comment:
`emplace_back` should also work (does it?) and save you move. Also check for the other tests

Reply:
it is just the tests. but fine